### PR TITLE
adjust validation to accomodate Ipv6 IP prefix

### PR DESCRIPTION
**Pull request by sm947**
## Description

Correcting prefix length validation logic for Ipv6 public ip prefix

## Checklist

- [ yes ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs [hunk 1]
```diff
@@ -138,9 +138,18 @@ private PSPublicIpPrefix CreatePublicIpPrefix()
                 publicIpPrefix.PublicIpAddressVersion = this.IpAddressVersion;
             }
 
+            if (publicIpPrefix.PublicIpAddressVersion == MNM.IPVersion.IPv6 && (this.PrefixLength < 117 || this.PrefixLength > 127))
+            {
+                throw new ArgumentException($"Cannot validate argument on parameter 'PrefixLength' {this.PrefixLength}. Supply an argument that is greater than or equal to 117 and less than or equal to 127 and then try the command again");
```
Comment (markcowl) [inline src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs]: This text should be in a resource string

File: src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs [hunk 2]
```diff
@@ -138,9 +138,18 @@ private PSPublicIpPrefix CreatePublicIpPrefix()
...
+            }
+            else if (publicIpPrefix.PublicIpAddressVersion == MNM.IPVersion.IPv4 && (this.PrefixLength < 21 || this.PrefixLength > 31))
+            {
+                throw new ArgumentException($"Cannot validate argument on parameter 'PrefixLength' {this.PrefixLength}. Supply an argument that is greater than or equal to 21 and less than or equal to 31 and then try the command again");
```
Comment (markcowl) [inline src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs]: same

File: src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs (hunk 1 above)
Comment (sm947) [reply src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs]: These two strings are different with different prefix lengths for IPv4 and IPv6. Can you please advise if I am missing something ?

**Comment by sm947** [comment]
> Can you add a change to the changelog as well?

Thank you. Added change information in changelog

File: src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs (hunk 1 above)
Comment (sm947) [reply src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs]: updated PR by using resource string

File: src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs (hunk 2 above)
Comment (sm947) [reply src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs]: Thank you. I have updated PR using resource string.

**Pull request merged**